### DISCUSSION TITLE
Question related to winget's --interactive flag

### DISCUSSION THREAD

**Discussion by felixsanz** [microsoft/winget-pkgs#131705] [Q&A] [answered]
When using `--interactive`, it brings up the installer setup screen so I can select what I want to install and what not (along other things like change install path). But what happens on upgrades? (`winget upgrade`), are this things remembered and respected?

**Comment by mdanish-kh** [comment]
`winget upgrade` simply invokes the installer with args specified in the manifest, and then it's the installer that controls how the current installation is updated. Currently winget doesn't store state, like installed location, of the application by itself. There are open issues for this feature request. Consider putting a 👍 to raise their priority:

- https://github.com/microsoft/winget-cli/issues/1002
- https://github.com/microsoft/winget-cli/issues/3045

**Reply by felixsanz** [reply]
so there is no point installing this packages with winget? because the upgrade mechanism will break your intentions on every release

**Reply by mdanish-kh** [reply]
If the installer supports a "Location" switch (i.e., specifying install location from the CLI args), then you can use `winget upgrade <package> --location <InstallLocation>` that will hopefully offer a graceful experience. If not, you can use the `--interactive` switch with the upgrade command to launch the installer and select a location from there

**Reply by felixsanz** [reply]
but what if I choose to install some components and some others not? i'll have to use `winget upgrade <package> --interactive` all the time to avoid having those unwanted components installed? so there is no upgrade mechanism here, it's a full install all the time?

**Reply by mdanish-kh** [reply, marked as answer]
For the packages that install multiple components, my answer would be similar to https://github.com/microsoft/winget-pkgs/discussions/131706#discussioncomment-7933442. WinGet cannot control what the installer will "install" or how the "upgrade" process will be carried out. It tries to offer a default experience as if you were running the installer manually with the silent switches. Any custom behavior needs to be specified via `--override` or `--custom` (or other switches e.g., `--location`) provided the installer has support for that functionality via the CLI args.
